src/data/finsupp.lean

```diff
@@ -1517,6 +1518,21 @@ variable (σ)
 lemma lt_wf : well_founded (@has_lt.lt (σ →₀ ℕ) _) :=
 subrelation.wf (sum_id_lt_of_lt) $ inv_image.wf _ nat.lt_wf
 
+@[simp] lemma nat_add_eq_zero (f g : α →₀ ℕ) :
```

Comment:
Wouldn't this hold in any `canonically_ordered_monoid`?

Reply:
No, I don't think so. I'm crucially relying on the fact that `0` is `⊥` in `ℕ`.

Reply:
This always holds, see `bot_eq_zero` in `ordered_group.lean`.

Reply:
Aha... I don't know anything about canonically ordered monoids (-;
I'll generalise.

Reply:
Done

Reply:
And you didn't even have to change anything to the proof! I would change the name of the lemma, though.

Reply:
Ha, good point. Done.